src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -303,6 +303,27 @@ void ZeroInferRequest::create_pipeline() {
     _logger.debug("ZeroInferRequest::create_pipeline - SyncInferRequest completed");
 }
 
+void ZeroInferRequest::check_and_update_batch_value(const ov::SoPtr<ov::ITensor>& tensor,
+                                                    const std::optional<size_t> batchSizeCandidate) {
+    if (batchSizeCandidate.has_value()) {
+        if (!_dynamicBatchValueChanged) {
+            if (tensor._ptr != nullptr && tensor->get_byte_size() != tensor->get_byte_size()) {
```

Comment:
`tensor->get_byte_size() != tensor->get_byte_size()`
Is this the right condition?